pkg/endpoint/local_endpoint.go

```diff
@@ -171,42 +174,67 @@ func GetLocalSpec(ctx context.Context, submSpec *types.SubmarinerSpecification,
 	}
 
 	endpointSpec := &submv1.EndpointSpec{
-		CableName:     fmt.Sprintf("submariner-cable-%s-%s", submSpec.ClusterID, strings.ReplaceAll(privateIP, ".", "-")),
 		ClusterID:     submSpec.ClusterID,
 		Hostname:      hostname,
-		PrivateIP:     privateIP,
 		NATEnabled:    submSpec.NATEnabled,
 		Subnets:       localSubnets,
 		Backend:       submSpec.CableDriver,
 		BackendConfig: backendConfig,
 	}
 
-	publicIP, resolver, err := getPublicIP(submSpec, k8sClient, backendConfig, airGappedDeployment)
-	if err != nil {
-		return nil, errors.Wrap(err, "could not determine public IP")
+	for _, privateIP := range privateIPs {
+		endpointSpec.SetPrivateIP(privateIP)
 	}
 
-	logger.Infof("Obtained local endpoint public IP %q using resolver %q", publicIP, resolver)
+	endpointSpec.CableName = fmt.Sprintf("submariner-cable-%s-%s", submSpec.ClusterID,
+		strings.ReplaceAll(endpointSpec.GetPrivateIP(k8snet.IPv4), ".", "-"))
+	publicIPs, err := getPublicIPs(submSpec, k8sClient, backendConfig, airGappedDeployment, true)
+
+	if len(publicIPs) == 0 && err != nil {
```

Comment:
Why is this checking `len(publicIPs)` here? If an error was returned with non-empty publicIPs then the code would proceed and silence the error. Perhaps the intention is to proceed if some IP was returned but I think it should fail if any errors occurs.